tests/testthat/_snaps/tree.md

```diff
@@ -0,0 +1,256 @@
+      │   ├─1:{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─2:{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      └─2:{list}
+        ├─id:"a"
+        └─val:2
+    Code
+      tree(deep_list, max_length = 0)
+      tree(deep_list, max_length = 2)
+    Output
+      [list]
+      ├─1:{list}
+    Code
+      tree(deep_list, max_depth = 1, max_length = 4)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+
+# Attributes are properly displayed as special children nodes
+
+    Code
+      list_w_attrs <- structure(list(structure(list(id = "a", val = 2), level = 2,
+      name = "first child"), structure(list(id = "b", val = 1, children = list(list(
+        id = "b1", val = 2.5))), level = 2, name = "second child", class = "custom-class"),
+      level = "1", name = "root"))
+      tree(list_w_attrs, show_attributes = TRUE)
+    Output
+      {list}
+      ├─{list}
+      │ ├─id:"a"
+      │ ├─val:2
+      │ ├┄<attr>names:"id","val"
+      │ ├┄<attr>level:2
+      │ └┄<attr>name:"first child"
+      ├─{custom-class}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ ├─children:[list]
+      │ ┊ └─1:{list}
+      │ ┊   ├─id:"b1"
+      │ ┊   ├─val:2.5
+      │ ┊   └┄<attr>names:"id","val"
+      │ ├┄<attr>names:"id","val","children"
+      │ ├┄<attr>level:2
+      │ ├┄<attr>name:"second child"
+      │ └┄<attr>class:"custom-class"
+      ├─level:"1"
+      ├─name:"root"
+      └┄<attr>names:"","","level","name"
+    Code
+      tree(list_w_attrs, show_attributes = FALSE)
+    Output
+      {list}
+      ├─{list}
+      │ ├─id:"a"
+      │ └─val:2
+      ├─{custom-class}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   └─1:{list}
+      │     ├─id:"b1"
+      │     └─val:2.5
+      ├─level:"1"
+      └─name:"root"
+
+# Handles elements with a single element and attributes well
+
+    Code
+      tree(list("first element", structure("second element", purpose = "show bug")),
+      show_attributes = TRUE)
+    Output
+      [list]
+      ├─1:"first element"
+      └─2:"second element"
+        └┄<attr>purpose:"show bug"
+
+# Large and multiline strings are handled gracefully
+
+    Code
+      long_strings <- list(`normal string` = "first element", `really long string` = paste(
+        rep(letters, 4), collapse = ""), `vec of long strings` = c(
+        "a long\nand multi\nline string element", "a fine length",
+        "another long\nand also multi\nline string element"))
+      tree(long_strings)
+    Output
+      {list}
+      ├─normal string:"first element"
+      ├─really long string:"abcdefghijklmnopqrstuvwxyzabcdef..."
+      └─vec of long strings:"a long and m...","a fine length","another long..."
+    Code
+      tree(long_strings, remove_newlines = FALSE)
+    Output
+      {list}
+      ├─normal string:"first element"
+      ├─really long string:"abcdefghijklmnopqrstuvwxyzabcdef..."
+      └─vec of long strings:"a long
```

Comment:
Should this show `\n` insteadd?

Reply:
I ended up using a unicode symbol to show where newlines existed instead of just wiping them out.
 ```
     {list}
      ├─normal string: "first element"
      ├─really long string: "abcdefghijklmnopqrstuvwxyzabcdef..."
      └─vec of long strings: ["a long↵and m...", "a fine length", "another long..."]
```
I figured it's less horizontal space taken up and a bit easier to parse on a glance. That being said I can see why we may want to simply use `\n` instead.

Reply:
Hmmmm, interesting. Again, I don't have any concerns except that we haven't used it before elsewhere.

Reply:
(This is another example where you want to balance the length on individual elements with the number of elements displayed)